code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -314,6 +337,49 @@
 		ADD_TRAIT(remaining_spiderling, TRAIT_IMMOBILE, WIDOW_ABILITY_TRAIT)
 	addtimer(CALLBACK(src, PROC_REF(grab_spiderlings), remaining_list, number_of_attempts_left - 1), 1)
 
+// ***************************************
+// *********** Cannibalise
+// ***************************************
+/datum/action/xeno_action/activable/cannibalise
+	name = "Cannibalise Spiderling"
+	ability_name = "Cannibalise Spiderling"
+	desc = "Consume one of your children, storing their biomass for future use. If any charges of Cannibalise are stored, alt-use of Birth Spiderling will create one spiderling in exchange for one charge of Cannibalise. Up to three charges of Cannibalise may be stored at once."
+	action_icon_state = "cannibalise_spiderling"
+	plasma_cost = 150
+	cooldown_timer = 2 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_CANNIBALISE_SPIDERLING,
+	)
+
+/datum/action/xeno_action/activable/cannibalise/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!owner.Adjacent(A))
+		owner.balloon_alert(owner, "Not adjacent")
+		return FALSE
+	if(!istype(A, /mob/living/carbon/xenomorph/spiderling))
+		owner.balloon_alert(owner, "We can't cannibalise this")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/cannibalise/use_ability(atom/A)
+	if(!do_after(owner, 0.5 SECONDS, TRUE, A, BUSY_ICON_DANGER))
+		return fail_activate()
+
+	owner.emote("roar")
+	var/mob/living/carbon/xenomorph/spiderling/to_cannibalise = A
+	to_cannibalise.death()
+	var/datum/action/xeno_action/create_spiderling/create_spiderling_action = owner.actions_by_path[/datum/action/xeno_action/create_spiderling]
```

Comment:
Not that it would ever matter in normal gameplay, but you should make sure that the action in question does exist before incrementing charges on it.